### PR TITLE
Oracle allow explicitly specifying vcn for subnet selection

### PR DESCRIPTION
**Pull request by a-dubs**
feat(oracle): ability to specify vcn by name for subnet selection

Oracle's pycloudlib implementation currently uses the most recently created VCN (Virtual Cloud Network) in the given compartment for new VM instances being created. Problems can therefore arise if a newly created VCN uses a subnet with different rules than the previously used VCN's subnet.

Now, instead of relying on just blindly choosing the most recently created VCN, pycloudlib supports specifying a vcn_name value which it will use to find a VCN that has a name that exactly matches the given vcn_name.

### CODE REVIEW

**Review by CalvoM — COMMENTED** [review]
LGTM, but I need to do some testing for the functionality.

File: pycloudlib/oci/utils.py [hunk 1]
```diff
@@ -47,13 +48,27 @@ def get_subnet_id(
         network_client: Instance of VirtualNetworkClient.
         compartment_id: Compartment where the subnet has to belong
         availability_domain: Domain to look for subnet id in.
+        target_subnet_name: Name of the subnet to look for.
```
Comment (holmanb) [inline pycloudlib/oci/utils.py:51]: Is this line needed?

**Review by CalvoM — APPROVED** [review]
Tested it with CTF and it works. LGTM.

File: pycloudlib/oci/utils.py [hunk 2]
```diff
@@ -37,6 +37,7 @@ def get_subnet_id(
     network_client: "oci.core.VirtualNetworkClient",  # type: ignore
     compartment_id: str,
     availability_domain: str,
+    vcn_name: str = None,
```
Comment (aciba90) [inline pycloudlib/oci/utils.py:40]:
```suggestion
    vcn_name: Optional[str] = None,
```

And the appropriate import if required.

**Review by aciba90 — CHANGES_REQUESTED** [review]
Thanks! Apart from the typing fix, LGTM.

File: pycloudlib/oci/utils.py (hunk 2 above)
Comment (a-dubs) [reply pycloudlib/oci/utils.py]: Fixed, tysm :D

**Comment by a-dubs** [comment]
Jira ticket for context: https://warthogs.atlassian.net/browse/CPC-2908

**Pull request merged**